Add `setupmeta` package.

matthewwardrop opened the pull request:
<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with
a PR or once it's ready for review, please let the right people know.
There are language-specific teams for reviewing recipes.

Currently available teams are:
- python `@conda-forge/help-python`
- python/c hybrid `@conda-forge/help-python-c`
- r `@conda-forge/help-r`
- java `@conda-forge/help-java`
- nodejs `@conda-forge/help-nodejs`
- c/c++ `@conda-forge/help-c-cpp`
- perl `@conda-forge/help-perl`
- Julia `@conda-forge/help-julia`
- ruby `@conda-forge/help-ruby`

If your PR doesn't fall into those categories please contact
the full review team `@conda-forge/staged-recipes`.

Due to GitHub limitations first time contributors to conda-forge are unable
to ping these teams. You can [ping the team](https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team) using a special command in
a comment on the PR to get the attention of the `staged-recipes` team. You can
also consider asking on our [Gitter channel](https://gitter.im/conda-forge/conda-forge.github.io)
or on our [Keybase chat](https://keybase.io/team/condaforge.chat)
if your recipe isn't reviewed promptly.
-->

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/setupmeta```) and found it was in an excellent condition.

matthewwardrop commented:
@zsimic As discussed elsewhere, for you to review :).

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/setupmeta```) and found some lint.

Here's what I've got...


For **recipes/setupmeta**:

 * The recipe must have some tests.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/setupmeta```) and found it was in an excellent condition.

zsimic commented:
I am confirming that I am willing to be listed here, and will gladly help should issues arise

matthewwardrop commented:
This is ready for review: @conda-forge/help-python.

matthewwardrop commented:
@conda-forge/help

matthewwardrop commented:
@conda-forge-admin, please ping team

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-webservice.

I was asked to ping @conda-forge/staged-recipes and so here I am doing that.

npavlovikj commented on recipes/setupmeta/LICENSE:
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Looks like the license file is already in the PyPI tarball, so this file can be removed from here.

npavlovikj commented:
@matthewwardrop , do you want to add yourself as a maintainer as well? I left a minor comment to remove the LICENSE file since it is included in the tarball - other than that the recipe looks good to me, so we can merge if afterwards.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/setupmeta```) and found some lint.

Here's what I've got...


For **recipes/setupmeta**:

 * license_file entry is missing, but is required.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/setupmeta```) and found it was in an excellent condition.

matthewwardrop commented:
@npavlovikj Thanks for the review. I've removed the LICENSE file, and updated meta.yaml accordingly (should I be adding the comment that the LICENSE file can now be found in the source tarball?). I also added myself as a maintainer, just in case :).

npavlovikj commented on recipes/setupmeta/meta.yaml:
```diff
@@ -0,0 +1,49 @@
+  home: "https://github.com/zsimic/setupmeta"
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE (inside source tarball)
```
Sorry if I wasn't clear earlier - please leave `license_file: LICENSE` here. The path to the license file, LICENSE, is relative to the source directory, which is the tarball, where the license is.

npavlovikj replied:
Also, please remove `(inside source tarball)` since that field should only contain the name of the license file.

npavlovikj commented:
Thank you for contributing @matthewwardrop 🎉

matthewwardrop commented:
And thanks for reviewing and helping, @npavlovikj !
